### ISSUE TITLE
Latest HDHomeRun Setup app blocked on Windows 10

### ISSUE THREAD

**Issue by frederick-bruckman** [mullvad/mullvadvpn-app#1564]
On Windows 10, Mullvad VPN 2020.03 blocks use of HDHomeRun devices after the hdhomerun 2020-02-05 update. The setup and config apps apps can't find the device, neither does already configured SageTV. This is with "allow LAN" in the Mullvad app, naturally. Turning off the kill switch does not help. It's the same with Wireguard and OpenVPN. The HDHomeRun device works fine with the VPN disconnected. It also works fine with OpenVPN GUI and the website-generated configs. It also worked with the 2019 HDHomeRun software.

My firewall app shows that UDP broadcasts from the private LAN interface to 255.255.255.255 are blocked (only) when the Mullvad app is connected. I don't have any more than that; I'm going to be using OpenVPN GUI unless/until this is fixed.

**Comment by pinkisemils** [comment]
Hiya, thank you for reporting the issue. We're trying to deduce whether a new whole has to be punched in the firewall, or if this is something that we broke with our recent (and critical) changes to our firewall - has this worked before with earlier versions of the application?

**Comment by frederick-bruckman** [comment]
Hi,

Thanks for the quick response. It "just worked" with earlier versions 
of both Mullvad and HDHomeRun software, but I seem to recall that 
either update would break it. I rolled back one (September 2019?) 
update to HDHomerun for that reason, but after the 2020.03 Mullvad 
app brought its own problems, I switched to OpenVPN GUI, and was 
then able to perform the very latest FW update for the device (Feb 
2020), and continue to use it with the VPN connected.

I'm currently using Malaware Byte's Windows Firewall Control (a 
front-end for Windows  Firewall), and blocking outgoing connections 
by default (with numerous exceptions) in lieu of a "kill switch". The 
HDHomeRun apps only need three rules to work with OpenVPN GUI: 
one to allow the app to connect on the "Public" interface to Any/443, 
and two to allow unrestricted access to the "Private" interface, one 
for in and one for out. WFC's log shows that the app first makes a 
connection on the external interface to the company's server @443 
(to check for FW updates or to spy on me perhaps ;) ), then the 
UDP broadcast from the private interface to 255.255.255.255@65001, 
then it ultimately makes a TCP connection from the private interface 
to the devices IP@65001.  The successful transaction doesn't log 
any incoming  connections; the response of the device to the UDP 
broadcast, where it reveals its IP address to the app, must be part 
of the initial outgoing "connection".

With Mullvad's app however, UDP packets from the private interface 
to 255.255.255.255@65001 show up in the list of blocked packets. 
(The official Wireguard app also fails, but it's not supposed to provide 
LAN access, so no surprise there.) There's no way to see what rule is 
actually blocking those packets, as far as I can tell. Why/how is the 
Mullvad app blocking those packets?

Fred
